interoperability_report.py

```diff
@@ -0,0 +1,878 @@
+        parameters,
+        testCase,
+        time_out,
+        producedCode,
+        samplesSent,
+        subscriber_finished, publisher_finished,
+        file,
+        verbosity):
+
+    """ Run the executable with the parameters and save
+        the error code obtained
+
+        name_executable     : name of the ShapeApplication to run
+                              as a Subscriber
+        parameters          : QOS to use with the Shape Application
+        testCase            : testCase is being tested
+                             (from rtps_test_suite_1)
+        time_out            : time pexpect waits until it finds a pattern
+        producedCode        : this variable will be overwritten with
+                              the obtained ReturnCode
+        samplesSent         : this variable contains the samples
+                              the Publisher sends
+        subscriber_finished : object event from multiprocessing that is set
+                              when the subscriber is finished
+        publisher_finished  : object event from multiprocessing that is set
+                              when the publisher is finished
+        file                : temporal file to save Shape Application output
+        verbosity           : print debug information
+
+        The function runs the Shape Application as a Subscriber
+        with the parameters defined.
+        The Subscriber Shape Application follows the next steps:
+            * The topic is created
+            * The Data Reader is created
+            * The Data Reader matches with a Data Writer
+            * The Data Reader detects the Data Writer as alive
+            * The Data Reader receives data
+
+        If the Shape Application achieves one step, it will print an specific
+        string pattern. The function will recognize that pattern and it will
+        continue also to the next step, waiting again for the next
+        corresponding pattern to be recognized. If the Shape Application
+        stops at some step, the function will not recognized the expected pattern
+        (or it will recognized an error pattern), it will save
+        the obtained ReturnCode and it will finish too.
+
+    """
+    # Step 1 : run the executable
+    log_message('Running Shape Application Subscriber', verbosity)
+    child_sub = pexpect.spawnu(f'{name_executable} {parameters}')
+    child_sub.logfile =  file
+
+    # Step 2 : Check if the topic is created
+    log_message('S: Waiting for topic creation', verbosity)
+    index = child_sub.expect(
+        [
+            'Create topic:',                                         # index = 0
+            pexpect.TIMEOUT,                                         # index = 1
+            'please specify topic name',                             # index = 2
+            'unrecognized value',                                    # index = 3
+            pexpect.EOF                                              # index = 4
```

Comment:
```suggestion
            'Create topic:', # index = 0
            pexpect.TIMEOUT, # index = 1
            'please specify topic name', # index = 2
            'unrecognized value', # index = 3
            pexpect.EOF # index = 4
```
I think it's easier to see to which line the comment refers if it doesn't contain spaces 😅